agent/agent.go

```diff
@@ -298,6 +298,44 @@ func (a *Agent) withSession(ctx context.Context, fn func(session *session) error
 	}
 }
 
+// AttachNetwork sends a container attach network request towards a
```

Comment:
We need the agent session because the server needs the node info for both authorization purposes as well as to get the node info where this request originated from (so that the task can be bound to that node). We don't need have that method here but then we need to export `withSession` so that the backend can use it.